src/gen.rs

```diff
@@ -17,38 +21,59 @@ pub struct Cmd {
 pub enum Output {
     /// Rust trait, client bindings, and test harness
     Rust,
+    /// Json representation of contract spec types
+    Json,
 }
 
 #[derive(thiserror::Error, Debug)]
 pub enum Error {
     #[error("generate rust from file: {0}")]
-    GenerateRustFromFile(rust::GenerateFromFileError),
+    GenerateRustFromFile(#[from] rust::GenerateFromFileError),
     #[error("format rust error: {0}")]
-    FormatRustError(syn::Error),
+    FormatRust(#[from] syn::Error),
+    #[error("host")]
+    Host(#[from] HostError),
+    #[error("xdr error: {0}")]
+    Xdr(#[from] soroban_env_host::xdr::Error),
+    #[error("io")]
+    Io(#[from] io::Error),
+    #[error("json error: {0}")]
+    Json(#[from] serde_json::Error),
+    #[error("contractnotfound")]
+    ContractSpecNotFound,
 }
 
 impl Cmd {
     pub fn run(&self) -> Result<(), Error> {
         match self.output {
             Output::Rust => self.generate_rust(),
+            Output::Json => self.generate_json(),
         }
     }
 
     pub fn generate_rust(&self) -> Result<(), Error> {
         let wasm_path_str = self.wasm.to_string_lossy();
-        let code =
-            rust::generate_from_file(&wasm_path_str, None).map_err(Error::GenerateRustFromFile)?;
+        let code = rust::generate_from_file(&wasm_path_str, None)?;
         let code_raw = code.to_string();
-        match syn::parse_file(&code_raw) {
-            Ok(file) => {
-                let code_fmt = prettyplease::unparse(&file);
-                println!("{}", code_fmt);
-                Ok(())
-            }
-            Err(e) => {
-                println!("{}", code_raw);
-                Err(Error::FormatRustError(e))
+        let code_fmt = prettyplease::unparse(&syn::parse_file(&code_raw)?);
```

Comment:
`syn::parse_file` is very finicky and will fail to parse in situations that are difficult to debug. This code intentionally printed out the raw code without formatting if this happens so we can debug it in those cases.

Reply:
I reverted all my changes to generate_rust